src/main/java/uk/ac/ebi/ena/webin/cli/WebinCliExecutor.java

```diff
@@ -316,4 +301,71 @@ private String getAuthTokenFromParam(){
     private boolean getTestModeFromParam(){
         return this.parameters.isTest();
     }
+
+    private File createSubmissionDir(String dir) {
+        if (StringUtils.isBlank(getSubmissionName())) {
+            throw WebinCliException.systemError(WebinCliMessage.EXECUTOR_INIT_ERROR.format("Missing submission name."));
+        }
+        File newDir = WebinCli.createOutputDir( parameters.getOutputDir(), String.valueOf( context ), getSubmissionName(), dir);
+        if (!FileUtils.emptyDirectory(newDir)) {
+            throw WebinCliException.systemError(WebinCliMessage.EXECUTOR_EMPTY_DIRECTORY_ERROR.format(newDir));
+        }
+        return newDir;
+    }
+
+    private String getManifestFileContent() {
+        try {
+            return new String(Files.readAllBytes(getParameters().getManifestFile().toPath()));
+        } catch (IOException ioe) {
+            throw WebinCliException.userError( "Exception thrown while reading manifest file", ioe.getMessage());
+        }
+    }
+
+    private String calculateManifestMd5() {
+        return FileUtils.calculateDigest( "MD5", parameters.getManifestFile());
+    }
+
+    private List<SubmissionBundle.SubmissionXMLFile> writeXmls(Map<SubmissionBundle.SubmissionXMLFileType, String> xmls) {
+        List<SubmissionBundle.SubmissionXMLFile> res = new ArrayList<>();
+
+        xmls.forEach((fileType, xml) -> {
+            String xmlFileName = fileType.name().toLowerCase() + ".xml";
+            Path xmlFilePath = getSubmitDir().toPath().resolve( xmlFileName );
+
+            try {
+                Files.write( xmlFilePath, xml.getBytes( StandardCharsets.UTF_8 ),
+                    StandardOpenOption.CREATE, StandardOpenOption.TRUNCATE_EXISTING, StandardOpenOption.SYNC );
+
+                res.add(new SubmissionBundle.SubmissionXMLFile(fileType, xmlFilePath.toFile(), null));
+            } catch(IOException ex) {
+                throw WebinCliException.systemError( ex );
+            }
+        });
+
+        res.add(createAllInOneSubmissionFile(xmls.values()));
+
+        return res;
+    }
+
+    private SubmissionBundle.SubmissionXMLFile createAllInOneSubmissionFile(Collection<String> xmls) {
+        Path xmlPath = getSubmitDir().toPath().resolve("aio-submission.xml");
+
+        try(BufferedWriter bw = new BufferedWriter(new FileWriter(xmlPath.toFile(), false))) {
+
+            bw.write("<WEBIN>" + System.lineSeparator());
+            for (String xml : xmls) {
+                bw.write(xml);
+                bw.write(System.lineSeparator());
+            }
+            bw.write("</WEBIN>");
+
+        } catch(IOException ex) {
+            throw WebinCliException.systemError( ex );
+        }
+
+        return new SubmissionBundle.SubmissionXMLFile(
+            SubmissionBundle.SubmissionXMLFileType.AIO_SUBMISSION,
+            xmlPath.toFile(),
+            FileUtils.calculateDigest( "MD5", xmlPath.toFile()));
+    }

```

Comment:
- I suggest that we create the final V2 submissions XML in SubmitService (more about this later).
- The SubmitService could treat the final V2 submissions XML as a temp file and not as a bundle file.